app/models/manageiq/providers/microsoft/infra_manager/refresh_parser.rb

```diff
@@ -413,6 +414,23 @@ def process_vm_hardware(vm)
       }
     end
 
+    def process_vm_custom_properties(vm)
+      result = []
+
+      custom_properties = vm['CustomProperty']
+      custom_properties.each do |key, value|
+        new_result = {
+          :section => 'custom_field',
+          :name    => key,
+          :value   => value,
+          :source  => 'VC'
+        }
+        result << new_result
```

Comment:
This could be simplified as:
```ruby
def process_vm_custom_properties(vm)
  vm['CustomProperty'].map do |key, value|
    {
      ...
    }
  end
end
```

Reply:
You probably also want to handle the case where this is nil, so `vm["CustomProperty"].to_a.map ...`